dali/test/python/test_external_source_parallel_shared_batch.py

```diff
@@ -116,22 +116,24 @@ def next_i():
         count += 1
         return count
     proc, task_queue, res_queue = setup_queue_and_worker(start_method, capacity, copy_callback, worker_params)
-    all_msgs = []
-    received = 0
-    for send_msg, recv_msg in zip(send_msgs, recv_msgs):
-        msgs = [ShmMessageDesc(next_i(), -next_i(), -next_i(), next_i(), -next_i()) for i in range(send_msg)]
-        all_msgs.extend(msgs)
-        _put_msgs(task_queue, msgs, send_one_by_one)
-        for _ in range(recv_msg):
-            [recv_msg] = res_queue.get()
-            msg_values = all_msgs[received].get_values()
-            received += 1
-            recv_msg_values = recv_msg.get_values()
-            assert len(msg_values) == len(recv_msg_values)
-            assert all(msg_value == recv_msg_value for msg_value, recv_msg_value in zip(msg_values, recv_msg_values))
-    if not proc.exitcode:
-        task_queue.close()
-    proc.join()
+    try:
```

Comment:
Q: no enter/exit to use `with proc`?

Reply:
For plain proc not really, but good point that it looked like a good candidate for ctx manager.